Foundation/FileManager.swift

```diff
@@ -509,28 +478,168 @@ open class FileManager : NSObject {
                 #elseif os(Linux) || os(Android) || CYGWIN
-                    let modeT = number.uint32Value
+                let modeT = number.uint32Value
                 #endif
-                _fileSystemRepresentation(withPath: path, {
+                try _fileSystemRepresentation(withPath: path, {
                     if chmod($0, mode_t(modeT)) != 0 {
-                        fatalError("errno \(errno)")
+                        throw _NSErrorWithErrno(errno, reading: false, path: path)
                     }
                 })
-            } else {
-                fatalError("Attribute type not implemented: \(attribute)")
+                
+            case .ownerAccountID:
+                guard let id = value as? NSNumber else {
+                    fatalError("Can't set file owner to \(value as Any?)")
+                }
+                try self._fileSystemRepresentation(withPath: path) {
+                    if chown($0, uid_t(id.uint32Value), 0 &- 1) != 0 {
+                        throw _NSErrorWithErrno(errno, reading: false, path: path)
+                    }
+                }
+                
+            case .ownerAccountName:
+                guard let name = value as? String else {
+                    fatalError("Can't set file owner name to \(value as Any?)")
+                }
+                try self._fileSystemRepresentation(withPath: path) {
+                    if let id = getpwnam(name)?.pointee.pw_uid {
+                        if chown($0, id, 0 &- 1) != 0 {
+                            throw _NSErrorWithErrno(errno, reading: false, path: path)
+                        }
+                    }
+                }
+                
+            case .groupOwnerAccountID:
+                guard let id = value as? NSNumber else {
+                    fatalError("Can't set file group owner to \(value as Any?)")
+                }
+                try self._fileSystemRepresentation(withPath: path) {
+                    if chown($0, 0 &- 1, uid_t(id.uint32Value)) != 0 {
+                        throw _NSErrorWithErrno(errno, reading: false, path: path)
+                    }
+                }
+                
+            case .groupOwnerAccountName:
+                guard let name = value as? String else {
+                    fatalError("Can't set file group owner name to \(value as Any?)")
+                }
+                try self._fileSystemRepresentation(withPath: path) {
+                    if let id = getgrnam(name)?.pointee.gr_gid {
+                        if chown($0, 0 &- 1, id) != 0 {
+                            throw _NSErrorWithErrno(errno, reading: false, path: path)
+                        }
+                    }
+                }
+                
+            case .immutable:
+                try NSURL(fileURLWithPath: path).setResourceValue(value, forKey: .isUserImmutableKey)
+                
+            case .busy, .extensionHidden:
+                NSUnimplemented()
```

Comment:
@parkera These keys are only meaningful on Darwin, should we just ignore these instead of crash.

Reply:
Hm, I think we should probably ignore them.

Reply:
Thanks. I will change these.